language/move-prover/stackless-bytecode-generator/src/function_target.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    annotations::Annotations,
+    stackless_bytecode::{AttrId, Bytecode},
+};
+use spec_lang::{
+    ast::Condition,
+    env::{FunId, FunctionEnv, Loc, TypeParameter},
+    symbol::{Symbol, SymbolPool},
+    ty::Type,
+};
+use std::collections::BTreeMap;
+use vm::file_format::CodeOffset;
+
+/// A FunctionTarget is a drop-in replacement for a FunctionEnv which allows to rewrite
+/// and analyze bytecode and parameter/local types. It encapsulates a FunctionEnv and information
+/// which can be rewritten using the `FunctionTargetsHolder` data structure.
+#[derive(Debug)]
+pub struct FunctionTarget<'env> {
+    pub func_env: &'env FunctionEnv<'env>,
+    pub data: &'env FunctionTargetData,
+}
+
+/// Holds the owned data belonging to a FunctionTarget, which can be rewritten using
```

Comment:
Based on the comment about owning, I expect the field `data`  to be a value rather than a reference.  Do you expect clients to want to share FunctionTargetData across different instances of FunctionTarget?

Reply:
The actual owner is FunctionTargetsHolder in function_target_pipeline. This object here is just kind of a handle. This allows to keep those things cached for frequent multiple usage.